pkg/provider/gitlab/gitlab.go

```diff
@@ -198,11 +198,14 @@ func (v *Provider) CreateStatus(_ context.Context, event *info.Event, statusOpts
 	body := fmt.Sprintf("**%s%s** has %s\n\n%s\n\n<small>Full log available [here](%s)</small>",
 		v.pacInfo.ApplicationName, onPr, statusOpts.Title, statusOpts.Text, detailsURL)
 
-	// in case we have access set the commit status, typically on MR from
-	// another users we won't have it but it would work on push or MR from a
-	// branch on the same repo or if token somehow can have access by other
-	// means.
-	// if we have an error fallback to send a issue comment
+	// In case we have access, set the commit status. Typically, on a Merge Request (MR)
+	// from a fork in an upstream repository, the token needs to have write access
+	// to the fork repository in order to create a status. However, the token set on the
+	// Repository CR usually doesn't have such broad access, preventing from creating
+	// a status commit on it.
```

Comment:
What is "status commit"? Do you mean status comment?

Reply:
actually @savitaashture I think in addition to the clarification here you need to add right before the 
```
//nolint: dogsled
```
comment below that is preceding `_, _, _ = v.Client.Commits.SetCommitStatus(event.SourceProjectID, event.SHA, opt)` an additional comment about ignoring errors because of the write access issues, along with some indication that we cannot reliable discern write access errors from other errors that we should process

Reply:
instead of the error ignore, as mentioned during pac call maybe we could p.eventEmitter in the user namespace an error that we cannot set status commit with the github token due of the err.Error() 

(and then you can remove the nolint dogsleg stuff)

Reply:
I like the emit event in user namespace idea / removing the nolint

I would still move the comment right above the error processing, but I'll defer to whatever you all want to do

Reply:
Addressed here : https://github.com/openshift-pipelines/pipelines-as-code/pull/1692/files#r1625635587